src/core/lib/iomgr/parse_address.cc

```diff
@@ -67,25 +67,54 @@ bool grpc_parse_unix_abstract(const grpc_uri* uri,
             uri->scheme);
     return false;
   }
+  grpc_error* error = grpc_fill_unix_abstract_name(uri->path, resolved_addr);
+  if (error != GRPC_ERROR_NONE) {
+    gpr_log(GPR_ERROR, "%s", grpc_error_string(error));
+    GRPC_ERROR_UNREF(error);
+    return false;
+  }
+  return true;
+}
+
+grpc_error* grpc_fill_unix_path(const absl::string_view path,
+                                grpc_resolved_address* resolved_addr) {
   struct sockaddr_un* un =
       reinterpret_cast<struct sockaddr_un*>(resolved_addr->addr);
   const size_t maxlen = sizeof(un->sun_path) - 1;
-  const size_t path_len = strnlen(uri->path, maxlen);
-  if (path_len == maxlen) return false;
+  if (path.size() > maxlen) {
+    return GRPC_ERROR_CREATE_FROM_COPIED_STRING(
+        absl::StrCat("Path name should not have more than ", maxlen,
+                     " characters")
+            .c_str());
+  }
+  un->sun_family = AF_UNIX;
+  path.copy(un->sun_path, path.size());
+  un->sun_path[path.size()] = '\0';
+  resolved_addr->len =
+      static_cast<socklen_t>(sizeof(un->sun_family) + path.size() + 1);
```

Comment:
In this case, shouldn't it just be `sizeof(*un)`?

Reply:
Are you confusing sizeof() and strlen()?

```
           struct sockaddr_un {
               sa_family_t sun_family;               /* AF_UNIX */
               char        sun_path[108];            /* Pathname */
           };
```

So sizeof(un) = sizeof(un->sun_family) + 108.

`#define AF_UNIX		1` and `sizeof(un->sun_family) = 2` so AF_UNIX is actually 0x001, which means `strlen(*un)` wouldn't work either.

Before, we were using sizeof(*un), which is _wrong_ according to the linux manual, but works anyway because the filesystem sees the null character and stops parsing.

Reply:
Oh, I missed the "at least" part

> The addrlen argument that describes the enclosing sockaddr_un
          structure should have a value of at least:

              offsetof(struct sockaddr_un, sun_path)+strlen(addr.sun_path)+1

Reply:
Done.